mapie/utils.py

```diff
@@ -404,3 +412,75 @@ def check_nan_in_aposteriori_prediction(X: ArrayLike) -> None:
             + "belongs to every resamplings.\n"
             "Increase the number of resamplings"
         )
+
+
+def check_lower_upper_bounds(
+    y_preds: NDArray,
+    y_pred_low: NDArray,
+    y_pred_up: NDArray
+) -> None:
+    """
+    Check if the lower or upper bounds are inconsistent.
+    If checking for MapieQuantileRegressor, then also checking
+    initial quantile values.
+
+    Parameters
+    ----------
+    y_pred : NDArray of shape (n_samples, 3) or (n_samples,)
+        All the predictions at quantile:
+        alpha/2, (1 - alpha/2), 0.5 or only the predictions
+    y_pred_low : NDArray of shape (n_samples,)
+        Final lower bound prediction with additional quantile
+        value added
+    y_pred_up : NDArray of shape (n_samples,)
+        Final upper bound prediction with additional quantile
+        value added
+    Raises
+    ------
+    Warning
+        If the aggregated predictions of any training sample would be nan.
+    Examples
+    --------
+    >>> import warnings
+    >>> warnings.filterwarnings("error")
+    >>> import numpy as np
+    >>> from mapie.utils import check_lower_upper_bounds
+    >>> y_preds = np.array([[4, 2, 3], [3, 4, 5], [2, 3, 4]])
+    >>> y_pred_low = np.array([4, 3, 2])
+    >>> y_pred_up = np.array([4, 4, 4])
+    >>> try:
+    ...     check_lower_upper_bounds(y_preds, y_pred_low, y_pred_up)
+    ... except Exception as exception:
+    ...     print(exception)
+    ...
+    WARNING: The initial prediction values from the quantile method
+    present issues as the upper quantile values might be higher than the
+    lower quantile values.
+    """
+    if y_preds.ndim == 1:
+        init_pred = y_preds
+    else:
+        init_pred, init_lower_bound, init_upper_bound = y_preds
+    if y_preds.ndim != 1 and np.any(np.logical_or(
+            init_lower_bound >= init_upper_bound,
+            init_pred <= init_lower_bound,
+            init_pred >= init_upper_bound
+            )
+    ):
+        warnings.warn(
+            "WARNING: The initial prediction values from the "
+            + "quantile method\npresent issues as the upper "
+            "quantile values might be higher than the\nlower "
+            + "quantile values."
+        )
```

Comment:
Add indentation and remove `y_preds.ndim != 1` for simplification ?

Reply:
The issue is that if we do that, then the coverage is not complete. I am not sure why this is occurring, but this was the quick fix I had found for this issue but it does have this "issue" of double check for `y_preds.ndim` unfortunately.

Reply:
Understood.